packages/@ngtools/benchmark/src/benchmark.ts

```diff
@@ -0,0 +1,113 @@
+import * as fs from 'fs';
+
+import { BenchmarkOptions } from './benchmark-options';
+import {
+  benchmarkLog,
+  isNumber,
+  average,
+  combine,
+  makeArray,
+  makeMatchFn,
+  matchSpawn,
+  serialMultiPromise
+} from './utils';
+
+export const benchmark = (benchmarkOptions: BenchmarkOptions) => {
+  const log = benchmarkLog.bind(null, benchmarkOptions);
+  const logIfDefined = (str: string, prop: any) => {
+    if (Array.isArray(prop) && prop.length === 0) {
+      prop = null;
+    }
+    return prop ? log(str + prop) : null;
+  };
+
+  const cwd = process.cwd();
+
+  // Build match function
+  let matchFn: any = null;
+  if (benchmarkOptions.match) {
+    matchFn = makeMatchFn(
+      benchmarkOptions.debug,
+      benchmarkOptions.match,
+      benchmarkOptions.matchCount,
+      benchmarkOptions.matchEditFile,
+      benchmarkOptions.matchEditString,
+    );
+  }
+
+  let editedFileContents: string;
+  if (benchmarkOptions.matchEditFile) {
+    // backup contents of file that is being edited for rebuilds
+    editedFileContents = fs.readFileSync(benchmarkOptions.matchEditFile, 'utf8');
+  }
+  // combine flags commands
+  let flagCombinations = combine(benchmarkOptions.extraArgs);
+  flagCombinations.unshift([]);
+
+  const startTime = Date.now();
+  log(`Base command: ${benchmarkOptions.command}`);
+  log(`Iterations: ${benchmarkOptions.iterations}`);
+  logIfDefined('Comment: ', benchmarkOptions.comment);
+  logIfDefined('Extra args: ', benchmarkOptions.extraArgs);
+  logIfDefined('Logging to: ', benchmarkOptions.logFile);
+  if (benchmarkOptions.match) {
+    log(`Match output: ${benchmarkOptions.match}`);
+    logIfDefined('Match count: ', benchmarkOptions.matchCount);
+    logIfDefined('Match edit file: ', benchmarkOptions.matchEditFile);
+    logIfDefined('Match edit string: ', benchmarkOptions.matchEditString);
+  }
+  if (benchmarkOptions.debug) {
+    log('### Debug mode, all output is logged ###');
+  }
+  log('');
+
+
+  let promise = Promise.resolve();
+  let hasFailures = false;
+
+  flagCombinations.forEach((flags) =>
+    promise = promise
+      .then(() => serialMultiPromise(
+        benchmarkOptions.iterations,
+        matchSpawn,
+        benchmarkOptions.debug,
+        cwd,
+        matchFn,
+        benchmarkOptions.command,
+        flags
+      ).then((results: any[]) => {
+        const failures = results.filter(result => result.err && result.err !== 0);
+        log(`Full command: ${benchmarkOptions.command} ${flags.join(' ')}`);
+
+        let times = results.filter(result => !result.err)
+          .map((result) => result.time);
+        log(`Time average: ${average(times)}`);
+        log(`Times: ${times.join()}`);
+
+        if (benchmarkOptions.match) {
+          let matches = results.filter(result => !result.err)
+            .map((result) => result.match);
+          if (matches.every(match => isNumber(match))) {
```

Comment:
You don't have to match on numbers. I haven't needed to match it myself, but don't see a reason to limit it to numbers.